homeassistant/components/synology_dsm/media_source.py

```diff
@@ -0,0 +1,239 @@
+"""Expose Synology DSM as a media source."""
+from __future__ import annotations
+
+import mimetypes
+from pathlib import Path
+import tempfile
+
+from aiohttp import web
+from synology_dsm.api.photos import SynoPhotosAlbum, SynoPhotosItem
+from synology_dsm.exceptions import SynologyDSMException
+
+from homeassistant.components import http
+from homeassistant.components.media_player import MediaClass
+from homeassistant.components.media_source import (
+    BrowseError,
+    BrowseMediaSource,
+    MediaSource,
+    MediaSourceItem,
+    PlayMedia,
+    Unresolvable,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .models import SynologyDSMData
+
+
+async def async_get_media_source(hass: HomeAssistant) -> MediaSource:
+    """Set up Synology media source."""
+    # Synology photos support only a single config entry
```

Comment:
This information is missing from the docs

Reply:
Multiple config entries are supported, they are passed into `SynologyPhotosMediaSource(hass, entries)`.
tbh this comment is confusing, so i'll remove it from code

Reply:
Thanks for the fix.